flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointNotifierITCase.java

```diff
@@ -447,5 +434,23 @@ public void notifyCheckpointComplete(long checkpointId) {
 				GeneratingSourceFunction.numPostFailureNotifications.incrementAndGet();
 			}
 		}
+
+		@Override
+		public List<Long> snapshotState(long checkpointId, long timestamp) throws Exception {
+			if (!hasFailed && count >= failurePos && getRuntimeContext().getIndexOfThisSubtask() == 0) {
```

Comment:
please move this methods up again to reduce the diff. The methods are identical apart from the signature and return statement, the diff should reflect that.